galera/src/ist.cpp

```diff
@@ -204,9 +216,98 @@ IST_determine_recv_addr (gu::Config& conf)
         recv_addr += ":" + gu::to_string(port);
     }
 
+    log_info << "IST receiver addr using " << recv_addr;
     return recv_addr;
 }
 
+static std::string IST_determine_recv_bind(gu::Config& conf) {
+	std::string recv_bind;
+
+	try {
+		recv_bind = conf.get(galera::ist::Receiver::RECV_BIND);
+	} catch (gu::NotFound&) {
+		// if not found take the value from RECV_ADDR
+		try {
+			recv_bind = conf.get(galera::ist::Receiver::RECV_ADDR);
+		} catch (gu::NotSet&) {
+			try {
+				recv_bind = conf.get(galera::BASE_HOST_KEY);
+			} catch (gu::NotSet&) {
+				gu_throw_error(EINVAL)<< "Could not determine IST receive bind address: '"
+				<< galera::ist::Receiver::RECV_BIND << "' not set.";
+			}
+		}
+	}
+	catch (gu::NotSet&)
+	{
+		// if not set take the value from RECV_ADDR
+		try
+		{
+			recv_bind = conf.get(galera::ist::Receiver::RECV_ADDR);
+		}
+		catch (gu::NotSet&)
+		{
+			try
+			{
+				recv_bind = conf.get(galera::BASE_HOST_KEY);
+			}
+			catch (gu::NotSet&)
+			{
+				gu_throw_error(EINVAL)
+				<< "Could not determine IST receive bind address: '"
+				<< galera::ist::Receiver::RECV_BIND << "' not set.";
+			}
+		}
+	}
+
+	/* check if explicit scheme is present */
+	if (recv_bind.find("://") == std::string::npos) {
+		bool ssl(false);
+
+		try {
+			std::string ssl_key = conf.get(gu::conf::ssl_key);
+			if (ssl_key.length() != 0)
+				ssl = true;
+		} catch (gu::NotSet&) {
+		}
+
+		if (ssl)
+			recv_bind.insert(0, "ssl://");
+		else
+			recv_bind.insert(0, "tcp://");
+	}
+
+	gu::URI rb_uri(recv_bind);
+
+	// Cargo-culting
+	// no idea what I'm doing
+	if (!conf.has(galera::BASE_HOST_KEY))
+		conf.set(galera::BASE_HOST_KEY, rb_uri.get_host());
```

Comment:
Thanks for the comment. What should be the strategy? Should I remove the "if"-block?